nncf/experimental/tensorflow/quantization/quantizers.py

```diff
@@ -0,0 +1,310 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Optional
+from typing import List
+
+import tensorflow as tf
+
+from nncf.common.utils.registry import Registry
+from nncf.common.quantization.structs import QuantizationMode
+from nncf.tensorflow.quantization.quantizers import TFQuantizerSpec
+from nncf.tensorflow.quantization.quantizers import Quantizer
+from nncf.tensorflow.layers.operation import InputType
+from nncf.tensorflow.quantization.functions import asymmetric_quantize
+from nncf.tensorflow.quantization.functions import symmetric_quantize
+from nncf.tensorflow.quantization.functions import asymmetric_range_initialization
+from nncf.tensorflow.quantization.functions import symmetric_range_initialization
+
+
+NNCF_QUANTIZATION_OPERATIONS_V2 = Registry('nncf_quantization_operations_v2')
+
+
+def _get_channel_size(input_shape: List[int], channel_axes: List[int]):
+    if not isinstance(channel_axes, (list, tuple)):
+        channel_axes = [channel_axes]
+    size = 1
+    for axis in channel_axes:
+        size *= input_shape[axis]
+    return size
+
+
+class QuantizerV2(Quantizer):
+    """
+    Base class for all quantization operations.
+    """
```

Comment:
Same comment about V2

Reply:
Thanks for the comment! I added the additional explanation to the docstring.